lib/inherited_resources.rb

```diff
@@ -25,6 +25,10 @@ module InheritedResources
   def self.flash_keys=(array)
     Responders::FlashResponder.flash_keys = array
   end
+
+  # Inherit from a different controller. This only has an effect if changed
+  # before InheritedResources::Base is loaded, e.g. in a rails initializer.
+  mattr_accessor(:parent_controller) { '::ApplicationController' }
```

Comment:
can I ask if this is the same as 

https://github.com/heartcombo/devise/blob/a259ff3c28912a27329727f4a3c2623d3f5cb6f2/lib/devise.rb#L244-L245

?

Ref: https://github.com/rails/rails/blob/56fbc632a378acc40d99f5b4be321f80a5d8b8c4/activesupport/lib/active_support/core_ext/module/attribute_accessors.rb#L208-L212

Reply:
@tagliala i'd say its equivalent – the block is evaluated and the class variable is set immediately: 

https://github.com/rails/rails/blob/56fbc632a378acc40d99f5b4be321f80a5d8b8c4/activesupport/lib/active_support/core_ext/module/attribute_accessors.rb#L69-L70

i think the devise code is just like that because when José Valim wrote these lines 13 years ago, mattr_accessor did not yet support passing a default value:

https://github.com/rails/rails/blob/bf526c2dbeb73bf11553004e43889a804b72866d/activesupport/lib/active_support/core_ext/module/attribute_accessors.rb#L60